### DISCUSSION TITLE
Feature request (adwaita): Keyboard navigation in tab overview

### DISCUSSION THREAD

**Discussion by FantomeBeignet** [ghostty-org/ghostty#4069] [Feature Requests, Ideas] [answered]
The tab overview is currently only really usable with the mouse (or enter/esc to go back to the focused tab), and basic keyboard navigation would greatly improve the usefulness of this feature. Here are a few things I have in mind:

- Tab/arrows navigation between tabs: currently, pressing the tab or arrow keys results in these keys being sent to the focused tab. Using them (as well as tab/shift-tab) to change the focused tab would be a great improvement. Having the `previous_tab`/`next_tab` working in the tab overview would also be great.
- Enter to select the only search result: currently, pressing enter when the tab search only has one results has no effect. Focusing the only search results would seem logical to me.
- Configurable navigation shortcuts: having the ability to set/modify navigation keybinds specific to the tab overview could be great, such as setting vim-like keybinds (ctrl+h/l) to navigate the tab overview. 

Although I have no Zig experience, I'd be happy to try and implement these features myself if they are deemed beginner-friendly enough.

**Comment by Norlock** [comment]
I would really love this as well, as it's one of the killer features but it should support KB navigation indeed. For me I think arrow or hjkl would be sufficient. I would like to use this instead of the search bar, because I normally don't have that many tabs open.

Maybe the search bar can be opened with a hotkey or something.

**Comment by pluiedev** [comment, marked as answer]
The tab overview is entirely implemented by [libadwaita](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.TabOverview.html) and I don't see an easy way to customize keybinds or even manually trigger the different elements inside the interface. Unfortunately it seems like this is something that libadwaita has to implement upstream.
